pkg/tanka/inline.go

```diff
@@ -45,38 +39,97 @@ func (i *InlineLoader) Load(path string, opts JsonnetOpts) (*v1alpha1.Environmen
 		return nil, fmt.Errorf("Found no environments in '%s'", path)
 	}
 
-	root, err := jpath.FindRoot(path)
+	// TODO: Re-serializing the entire env here. This is horribly inefficient
+	envData, err := json.Marshal(envs[0])
 	if err != nil {
 		return nil, err
 	}
 
-	file, err := jpath.Entrypoint(path)
+	env, err := inlineParse(path, envData)
 	if err != nil {
 		return nil, err
 	}
 
-	namespace, err := filepath.Rel(root, file)
+	return env, nil
+}
+
+func (i *InlineLoader) Peek(path string, opts LoaderOpts) (*v1alpha1.Environment, error) {
+	opts.JsonnetOpts.EvalScript = EnvsOnlyEvalScript
+	if opts.Name != "" {
+		opts.JsonnetOpts.EvalScript = fmt.Sprintf(SingleEnvEvalScript, opts.Name)
+	}
+	return i.Load(path, opts)
+}
+
+func (i *InlineLoader) List(path string, opts LoaderOpts) ([]*v1alpha1.Environment, error) {
+	opts.JsonnetOpts.EvalScript = EnvsOnlyEvalScript
+	if opts.Name != "" {
+		opts.JsonnetOpts.EvalScript = fmt.Sprintf(SingleEnvEvalScript, opts.Name)
+	}
```

Comment:
This behavior doesn't appear to be correct to me – `List` is supposed to give all environments that could possibly be loaded from that path. If you already know what your environment is called, use `Peek` or `Load` instead